docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -0,0 +1,125 @@
+# Multi-Tenancy Network
+
+Xiaoning Ding, Hong Chen
+
+## Introduction
+
+This document is a part of the multi-tenancy design documents in Arktos. It describes network-related design changes to support multi-tenancy, including network model, backward-compatibility, DNS and service changes, etc.
+
+Please refer to other design documents in the "multi-tenancy" document directory if you want to learn about other parts in the multi-tenancy design.
+
+## Network Isolation
+
+Network isolation is achieved with a new API object: network:
+
+* Every pod is associated with a certain network object. If not specified it will be associated to a default network object, which is created when a space is initialized. 
+* A pod can only communicate with pods within the same network.
+* Every network has its own DNS server. The DNS server pod is automatically created when a new network object is created.
```

Comment:
I see. It makes sense. There are some details to be figured out here about when to deploy the actual pods. Since a design document should be clear about design decisions, I didn't put the "usually...." sentence and instead I put a TBD there for the design decision:

"* Every network has its own DNS service. An associated DNS service is automatically created when a new network object is created. (**tbd: decide when to deploy the actual DNS server pods**)"